src/main/java/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,15 @@
+package dao;
+
+import db.Storage;
+
+public class FruitDaoImpl implements FruitDao {
+    @Override
+    public void add(String fruitName, Integer quantity) {
+        Storage.fruits.put(fruitName, quantity);
```

Comment:
According to the encapsulation principle, you should avoid direct manipulation of the storage map from outside of the Storage class. Consider adding methods to the Storage class that encapsulate this logic, such as `addFruit` or `updateQuantity`.